### PR TITLE
Improve `simulate_input`

### PR DESCRIPTION
**Pull request by CryZe**
Various common operations were not implemented. I'm not 100% why these weren't implemented, but I'm assuming this is because the IME work isn't done yet.

### CODE REVIEW

File: druid-shell/src/text.rs [hunk 1]
```diff
@@ -531,6 +549,38 @@ pub fn simulate_input<H: WinHandler + ?Sized>(
...
             input_handler.handle_action(action);
         }
+        KbKey::Home => {
+            let movement = Movement::ParagraphStart;
+            if event.mods.shift() {
+                input_handler.handle_action(Action::MoveSelecting(movement));
+            } else {
+                input_handler.handle_action(Action::Move(movement));
+            }
+        }
```
Comment (Maan2003) [inline druid-shell/src/text.rs:559]: this doesn't looks correct to me, home moves to the start of line, and ctrl-home to the start of document. 
(similarly for end)

File: druid-shell/src/text.rs [hunk 2]
```diff
@@ -531,6 +549,38 @@ pub fn simulate_input<H: WinHandler + ?Sized>(
...
+        KbKey::End => {
+            let movement = Movement::ParagraphEnd;
+            if event.mods.shift() {
+                input_handler.handle_action(Action::MoveSelecting(movement));
+            } else {
+                input_handler.handle_action(Action::Move(movement));
+            }
+        }
+        KbKey::PageUp => {
+            let movement = Movement::Vertical(VerticalMovement::DocumentStart);
```
Comment (Maan2003) [inline druid-shell/src/text.rs:569]: `VerticalMovement::PageUp` looks like a better candidate

**Comment by CryZe** [comment]
Alright I made the changes. However it seems like page up and down movement simply isn't implemented in the textbox for now, but I guess that's fine for now.

**Comment by CryZe** [comment]
Seems like the CI is trolling me.

**Review by Maan2003 — APPROVED** [review]
Thanks!

**Comment by cmyr** [comment]
Yea pageup/pagedown aren't implemented because they're surprisingly tricky; they don't modify the selection or cursor at all, and are instead handled by the scrollview.

**Pull request merged**